File: chantier-manager-api/src/main/java/istic/mmm/chantier_manager_api/entities/PhotoChantier.java
package istic.mmm.chantier_manager_api.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "photos_chantier")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PhotoChantier {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String url; // ou chemin local

    private LocalDateTime datePrise;

    private String commentaire;

    // Photo liée à un chantier (vue générale)
    @ManyToOne
    @JoinColumn(name = "chantier_id")
    private Chantier chantier;

    // Optionnellement liée à une anomalie précise
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "anomalie_id")
    @JsonIgnore
    private Anomalie anomalie;
}

File: chantier_manager_mobile/lib/models/photo_chantier.dart
class PhotoChantier {
  final int id;
  final String url;
  final DateTime? datePrise;
  final String? commentaire;
  final int? chantierId;
  final int? anomalieId;

  const PhotoChantier({
    required this.id,
    required this.url,
    this.datePrise,
    this.commentaire,
    this.chantierId,
    this.anomalieId,
  });

  factory PhotoChantier.fromJson(Map<String, dynamic> json) {
    return PhotoChantier(
      id: json['id'] as int,
      url: json['url'] as String? ?? '',
      datePrise: json['datePrise'] != null
          ? DateTime.tryParse(json['datePrise'] as String)
          : null,
      commentaire: json['commentaire'] as String?,
      chantierId: json['chantierId'] as int?,
      anomalieId: json['anomalieId'] as int?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'url': url,
      'datePrise': datePrise?.toIso8601String(),
      'commentaire': commentaire,
      'chantierId': chantierId,
      'anomalieId': anomalieId,
    };
  }
}
